Card/src/main/java/com/braintreepayments/api/card/BaseCard.kt

```diff
@@ -0,0 +1,116 @@
+package com.braintreepayments.api.card
+
+import androidx.annotation.RestrictTo
+import com.braintreepayments.api.core.PaymentMethod
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Base class used to build various types of cards
+ */
+abstract class BaseCard(
+    open val cardholderName: String? = null,
+    open val number: String? = null,
+    open val company: String? = null,
+    open val countryCode: String? = null,
+    open val cvv: String? = null,
+    open var expirationMonth: String? = null,
+    open var expirationYear: String? = null,
+    open val extendedAddress: String? = null,
+    open val firstName: String? = null,
+    open val lastName: String? = null,
+    open val locality: String? = null,
+    open val postalCode: String? = null,
+    open val region: String? = null,
+    open val streetAddress: String? = null,
```

Comment:
Should we make all of these `val`s `var`s instead? This way Java integrations are able to call the generated setter functions.

Reply:
Thank you for catching the error, this will be addressed after removing super classes from Card.